process/block/preprocess/transactionsV2.go

```diff
@@ -0,0 +1,416 @@
+package preprocess
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage/txcache"
+)
+
+func (txs *transactions) createAndProcessMiniBlocksFromMeV2(
+	haveTime func() bool,
+	isShardStuck func(uint32) bool,
+	isMaxBlockSizeReached func(int, int) bool,
+	sortedTxs []*txcache.WrappedTransaction,
+) (block.MiniBlockSlice, error) {
+	log.Debug("createAndProcessMiniBlocksFromMeV2 has been started")
+
+	mapMiniBlocks := make(map[uint32]*block.MiniBlock)
+	mapSCTxs := make(map[string]struct{})
+	mapTxsForShard := make(map[uint32]int)
+	mapScsForShard := make(map[uint32]int)
+
+	numTxsAdded := 0
```

Comment:
you could make a structure for all these num* variables

Reply:
Done